### PR TITLE
[chore] move googlecloud exporter to generated lifecycle tests

### PR DESCRIPTION
**Pull request by Frapschen**
Relates to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27849

After enabled `skip_lifecycle`, still get these error:
```
=== RUN   TestComponentLifecycle
=== RUN   TestComponentLifecycle/logs-shutdown
    generated_component_test.go:82: 
        	Error Trace:	/Users/fraps/daocloud/github-code/opentelemetry-collector-contrib/exporter/googlecloudexporter/generated_component_test.go:82
        	Error:      	Received unexpected error:
        	            	error finding default application credentials: google: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information
        	Test:       	TestComponentLifecycle/logs-shutdown
=== RUN   TestComponentLifecycle/logs-lifecycle
=== RUN   TestComponentLifecycle/metrics-shutdown
    generated_component_test.go:82: 
        	Error Trace:	/Users/fraps/daocloud/github-code/opentelemetry-collector-contrib/exporter/googlecloudexporter/generated_component_test.go:82
        	Error:      	Received unexpected error:
        	            	error finding default application credentials: google: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information
        	Test:       	TestComponentLifecycle/metrics-shutdown
=== RUN   TestComponentLifecycle/metrics-lifecycle
=== RUN   TestComponentLifecycle/traces-shutdown
    generated_component_test.go:82: 
        	Error Trace:	/Users/fraps/daocloud/github-code/opentelemetry-collector-contrib/exporter/googlecloudexporter/generated_component_test.go:82
        	Error:      	Received unexpected error:
        	            	error finding default application credentials: google: could not find default credentials. See https://cloud.google.com/docs/authentication/external/set-up-adc for more information
        	Test:       	TestComponentLifecycle/traces-shutdown

```

### CODE REVIEW

File: exporter/googlecloudexporter/metadata.yaml [hunk 1]
```diff
@@ -7,3 +7,8 @@ status:
   distributions: [contrib, observiq]
   codeowners:
     active: [aabmass, dashpole, jsuereth, punya, damemi, psx95]
+
```
Comment (crobert-1) [inline exporter/googlecloudexporter/metadata.yaml:10]:
```suggestion
```

Just a nit, one empty line is probably fine here 👍

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
@Frapschen can you please look into CI failures and rebase?

**Comment by dashpole** [comment]
I don't think the google cloud exporter will work with these tests as-is, since we create our client during creation, rather than during Start().

**Comment by dashpole** [comment]
It is on my list of things to do.

**Pull request closed without merging**